internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -251,11 +255,18 @@ func (m DeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 				}, false),
 		},
 
-		"configuration": {
-			Type:     pluginsdk.TypeList,
-			Optional: true,
-			Computed: true,
-			MaxItems: 1,
+		"configuration": {},
+
+		"tags": commonschema.Tags(),
+	}
+
+	if !features.FourPointOhBeta() {
```

Comment:
is this how features are deprecated in TF? I have not looked at this function but I vaguely remember that this was for breaking changes when the provider gets a 4.0 bumped.

Reply:
Good question! I got that from this contributing guide: https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-new-fields-to-resource.md#renaming-and-deprecating-a-property, but could use some direction here if there's a different way.

Reply:
Thanks for sharing this 👍